src/main/asciidoc/index.adoc

```diff
@@ -17,503 +17,801 @@
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Although originally designed for third-party apps, OAuth 2.0 SHOULD nowadays be used even for first-party apps, as
+sharing credentials with anything other than the centralized - and usually highly-secure and well-audited -
+authorization server is considered an anti-pattern (and is therefore NOT RECOMMENDED).
+====
+
+Propelled by its adoption by Facebook and Twitter, OAuth 2.0 quickly became widely popular.
+
+This popularity seemingly blinded many people to the many challenges of OAuth 2.0.
+
+For starters, the specification is - regrettably - not standardized, plagued by an abundance of options and a
+frustrating lack of constraints leading to numerous insecure and non-interoperable implementations.
+
+What's worse, many important topics (e.g., client registration, authentication, authorization server capabilities) are
+simply not part of the core specification and were only addressed - partially - through the subsequent few hundred pages
+of "companion" specifications (e.g., https://datatracker.ietf.org/doc/html/rfc6750[RFC 6750],
+https://datatracker.ietf.org/doc/html/rfc6819[RFC 6819]).
+
+Thus, authentication, which is an important prerequisite for authorization, is, to date, simply considered the
+responsibility of the authorization server.
+
+Nevertheless, it's not all bad and, if implemented correctly, OAuth 2.0 is a vast improvement on what was being done in
+the past.
+
+OAuth 2.0 defines four roles:
+
+* the resource owner: an entity (end-user or server) capable of granting access to a protected resource;
+* the resource server: a server (e.g., REST API) hosting the protected resource;
+* the client: an application requesting access to a protected resource on behalf of the resource owner;
+* the authorization server: a server issuing tokens to the client after successfully authenticating the resource owner
+and obtaining authorization.
+
+The authorization process of OAuth 2.0 utilizes two authorization server endpoints:
+
+* the authorization endpoint: an endpoint used by the client to obtain authorization from the resource owner;
+* the token endpoint: an endpoint used by the client to exchange an authorization grant for an access token.
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+To access a protected resource, the client MUST pass the access token in the Authorization header of the request using
+the Bearer HTTP authentication scheme (defined in RFC 6750 as part of the HTTP authentication framework).
+====
 
-* Prepare your HTTP message using the JWS.
-** The access tokens in OAuth 2.0 are of type *bearer*, meaning the client needs to pass the token with each request.
-** The followings are the expected HTTP headers:
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The authorization server MUST be configured so that it returns access tokens that are self-contained in lieu of opaque
```

Comment:
Since scopes should be fairly restrictive, a client should not be able to use a token on more than one API (unless the token is propagated in a microservices architecture).

Reply:
Create an issue to elaborate on this issue.